src/select/Select.ts

```diff
@@ -0,0 +1,506 @@
+ *
+ * @cssprop --omni-select-scrollbar-track-border-radius - Select items scroll bar track border radius.
+ * @cssprop --omni-select-scrollbar-track-background-color - Select items scroll bar track background color.
+ *
+ * @cssprop --omni-select-scrollbar-thumb-border-radius - Select items scroll bar thumb border radius.
+ * @cssprop --omni-select-scrollbar-thumb-background-color - Select items scroll bar thumb background color.
+ * @cssprop --omni-select-scrollbar-track-padding-top - Select items scroll bar thumb top padding.
+ * @cssprop --omni-select-scrollbar-track-padding-bottom - Select items scroll bar thumb bottom padding.
+ * @cssprop --omni-select-scrollbar-track-padding-left - Select items scroll bar thumb left padding.
+ * @cssprop --omni-select-scrollbar-track-padding-right - Select items scroll bar thumb right padding.
+ *
+ * @cssprop --omni-select-item-font-color - Select item font color.
+ * @cssprop --omni-select-item-font-family - Select item font family.
+ * @cssprop --omni-select-item-font-weight - Select item font weight.
+ * @cssprop --omni-select-item-padding-top - Select item top padding.
+ * @cssprop --omni-select-item-padding-bottom - Select item bottom padding.
+ * @cssprop --omni-select-item-padding-left - Select item left padding.
+ * @cssprop --omni-select-item-padding-right - Select item right padding.
+ * @cssprop --omni-select-item-width - Select item width.
+ *
+ * @cssprop --omni-select-item-hover-background-color - Select item hover background color.
+ * @cssprop --omni-select-item-selected-color - Selected item color.
+ * @cssprop --omni-select-item-none-hover - Select item hover.
+ *
+ * @cssprop --omni-select-loading-indicator-width - Select loading indicator width.
+ * @cssprop --omni-select-loading-indicator-height - Select loading indicator height.
+ *
+ */
+@customElement('omni-select')
+export class Select extends OmniFormElement {
+    @query('#select')
+    private _selectElement: HTMLInputElement;
+
+    /**
+     * Selectable items of the select component.
+     * @attr
+     */
+    @property({ type: Array, reflect: true }) items: SelectItems | (() => SelectItems);
+
+    /**
+     * The field of the item to display as one of the selectable options.
+     * @attr
+     */
+    @property({ type: String, reflect: true, attribute: 'display-field' }) displayField: string;
+
+    /**
+     * The id field of the items provided.
+     * @attr
+     */
+    @property({ type: String, reflect: true, attribute: 'id-field' }) idField: string = 'id';
+
+    /**
+     * The render function for each item.
+     */
+    @property({ type: Object, reflect: false }) renderItem: RenderFunction;
+
+    // Internal state properties
+    @state() private _popUp: boolean = false;
+    @state() private _bottomOfScreen: boolean = false;
+    @state() private _mobileWidth: boolean = false;
+
+    constructor() {
+        super();
+        this._sizeCheck();
```

Comment:
Should this not be done in connectedCallback? Or perhaps firstUpdated?

Reply:
Adding to connectedCallback